crates/bevy_math/src/bounding/bounded2d/primitive_impls.rs

```diff
@@ -0,0 +1,411 @@
+        BoundingCircle::new(translation, f32::MAX / 2.0)
+    }
+}
+
+impl Bounded2d for Line2d {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        let direction = rotate_vec2(*self.direction, rotation);
+
+        // Dividing `f32::MAX` by 2.0 can actually be good so that we can do operations
+        // like growing or shrinking the AABB without breaking things.
+        let max = f32::MAX / 2.0;
+        let half_width = if direction.x == 0.0 { 0.0 } else { max };
+        let half_height = if direction.y == 0.0 { 0.0 } else { max };
+        let half_size = Vec2::new(half_width, half_height);
+
+        Aabb2d {
+            min: translation - half_size,
+            max: translation + half_size,
+        }
+    }
+
+    fn bounding_circle(&self, translation: Vec2, _rotation: f32) -> BoundingCircle {
+        BoundingCircle::new(translation, f32::MAX / 2.0)
+    }
+}
+
+impl Bounded2d for Segment2d {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        // Rotate the segment by `rotation`
+        let direction = Direction2d::from_normalized(rotate_vec2(*self.direction, rotation));
+        let segment = Self { direction, ..*self };
+        let (point1, point2) = (segment.point1(), segment.point2());
+
+        Aabb2d {
+            min: translation + point1.min(point2),
+            max: translation + point1.max(point2),
+        }
+    }
+
+    fn bounding_circle(&self, translation: Vec2, _rotation: f32) -> BoundingCircle {
+        BoundingCircle::new(translation, self.half_length)
+    }
+}
+
+impl<const N: usize> Bounded2d for Polyline2d<N> {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        Aabb2d::from_point_cloud(translation, rotation, self.vertices)
+    }
+
+    fn bounding_circle(&self, translation: Vec2, rotation: f32) -> BoundingCircle {
+        BoundingCircle::from_point_cloud(translation, rotation, &self.vertices)
+    }
+}
+
+impl Bounded2d for BoxedPolyline2d {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        Aabb2d::from_point_cloud(translation, rotation, self.vertices.to_vec())
+    }
+
+    fn bounding_circle(&self, translation: Vec2, rotation: f32) -> BoundingCircle {
+        BoundingCircle::from_point_cloud(translation, rotation, &self.vertices)
+    }
+}
+
+impl Bounded2d for Triangle2d {
+    fn aabb_2d(&self, translation: Vec2, rotation: f32) -> Aabb2d {
+        let [a, b, c] = self.vertices.map(|vtx| rotate_vec2(vtx, rotation));
+
+        let min = Vec2::new(a.x.min(b.x).min(c.x), a.y.min(b.y).min(c.y));
+        let max = Vec2::new(a.x.max(b.x).max(c.x), a.y.max(b.y).max(c.y));
+
+        Aabb2d {
+            min: min + translation,
+            max: max + translation,
+        }
+    }
+
+    fn bounding_circle(&self, translation: Vec2, rotation: f32) -> BoundingCircle {
+        let (Circle { radius }, circumcenter) = self.circumcircle();
```

Comment:
I see an issue with using the circumcircle.

In most real cases, the smallest enclosing circle is in fact, the circumcircle.
However, that is not true for obtuse triangles, there the smallest enclosing circle is actually determined by the longest side.

The problem is that given very obtuse triangles, the circumcircle can get arbitrarily large, leading to very (badly) surprising results.

I'd propose to check for the triangle being obtuse. To do this one should be able to just take the scalar products of the adjacent line segment vectors, if the angle is acute, the scalar product will be positive, if the angle is a right angle or is obtuse, the scalar product will be zero or negative.
One then computes the bounding circle for the line segment opposite to the right/obtuse angle.
(In the case of a right-angle triangle, the circumcircle is the same as this bounding circle, in the case of an obtuse angle, this bounding circle is smaller.)

Reply:
Oh yeah, I should mention that the vectors need to have same orientation in the method I described. Say you're computing the angle at vertex A, then both vectors need to be either (B-A) dot (C-A) or (A-B) dot (A-C), combining these alternatives will give results with an opposite sign (workable, but needs to be taken care of).

Reply:
Does this look fine?
```rust
fn bounding_circle(&self, translation: Vec2, rotation: f32) -> BoundingCircle {
    let [a, b, c] = self.vertices;

    // The points of the segment opposite to the obtuse or right angle if one exists
    let side_opposite_to_non_acute = if (b - a).dot(c - a) <= 0.0 {
        Some((b, c))
    } else if (c - b).dot(a - b) <= 0.0 {
        Some((c, a))
    } else if (a - c).dot(b - c) <= 0.0 {
        Some((a, b))
    } else {
        // The triangle is acute.
        None
    };

    if let Some((point1, point2)) = side_opposite_to_non_acute {
        // The triangle is obtuse or right, so the minimum bounding circle's diameter is equal to the longest side.
        // We can compute the minimum bounding circle from the line segment of the longest side.
        let (segment, center) = Segment2d::from_points(point1, point2);
        segment.bounding_circle(rotate_vec2(center, rotation) + translation, rotation)
    } else {
        // The triangle is acute, so the smallest bounding circle passes through all three vertices,
        // forming the circumcircle of the triangle.
        let (Circle { radius }, circumcenter) = self.circumcircle();
        BoundingCircle::new(rotate_vec2(circumcenter, rotation) + translation, radius)
    }
}
```
It looks like it might be working:
![kuva](https://github.com/bevyengine/bevy/assets/57632562/1887cf4c-eb46-46fe-9e6e-1d3e08a1a176)

Reply:
Slightly unrelated thought: In a future PR, I think it'd also be cool to add an enum like `TriangleKind` for determining whether it's right, acute, or obtuse. Although it'd also be nice to have something similar for equilateral, isosceles, and scalene triangles, so perhaps separate enums like `TriangleAngleKind` and `TriangleSideKind` could be better

Reply:
That looks good to me!
The comments are very good.

Reply:
Implemented this for triangles and cones now. The conical frustum still uses a circumcircle though because the same logic probably wouldn't work for trapezoids.